browser-test/src/applicant/application_version_fast_forward_bulkstatus.test.ts

```diff
@@ -0,0 +1,2052 @@
+    expect(id).toBeGreaterThan(0)
+
+    return id
+  }
+
+  /**
+   * Gets the program id from the application review page url
+   * @returns {number} programId. If the ID is not found a failed assertion will end the test.
+   */
+  getProgramIdFromApplicationReviewUrl(): number {
+    const pattern = /\/programs\/(?<id>[0-9]*)\/*/
+    const id = parseInt(this.page.url().match(pattern)?.groups?.id || '-1')
+    expect(id).toBeGreaterThan(0)
+
+    return id
+  }
+
+  /**
+   * Waits for the program disabled page to load
+   */
+  async waitForDisabledPage(): Promise<void> {
+    await this.page.waitForURL(`**/programs/${this.programName}/disabled`)
+    await expect(
+      this.page.getByRole('heading', {name: 'This program is no longer'}),
+    ).toBeVisible()
+  }
+}
+
+/**
+ * This class maintains the state and logic used by the Program Admin actor
+ */
+class FastForwardProgramAdminActor {
+  private programName: string
+  private page: Page
+  private adminPrograms: AdminPrograms
+
+  /**
+   * @constructor
+   * @param {string} programName name of the program
+   * @param {Page} page Unique page instance for this actor
+   */
+  private constructor(programName: string, page: Page) {
+    this.programName = programName
+    this.page = page
+    this.adminPrograms = new AdminPrograms(page)
+  }
+
+  /**
+   * Simplifies creation of the {FastForwardProgramAdminActor}
+   * @param {string} programName
+   * @param {Browser} browser instance from Playwright
+   * @returns {Promise<FastForwardProgramAdminActor>} Promise to an instance of this class
+   */
+  public static async create(
+    programName: string,
+    browser: Browser,
+  ): Promise<FastForwardProgramAdminActor> {
+    const context = await browser.newContext({
+      recordVideo: {
+        dir: 'tmp/videos/',
+      },
+    })
+    return new FastForwardProgramAdminActor(
+      programName,
+      await context.newPage(),
+    )
+  }
+
+  /**
+   * Get the playwright page object bound to this actor
+   */
+  getPage(): Page {
+    return this.page
+  }
+
+  /**
+   * Close to cleanup at the end of the test
+   */
+  async closeBrowserContext() {
+    await this.page.context().close()
+  }
+
+  /**
+   * Log in the program admin actor
+   */
+  async login() {
+    await this.page.goto('/programs')
+    await closeWarningMessage(this.page)
+    await loginAsProgramAdmin(this.page)
+  }
+
+  /**
+   * Log out the program admin actor
+   */
+  async logout() {
+    await logout(this.page)
+  }
+
+  /**
+   * Navigate to the view applications page at `/admin/programs/:programId/applications
+   */
+  async viewApplications() {
+    await this.adminPrograms.viewApplications(this.programName)
+  }
+
+  /**
+   * Get a locator any admin application rows. There may be 0-♾️
+   * @returns {Locator} Locator to admin application row
+   */
+  getRowLocator(): Locator {
+    return this.page.locator('.cf-admin-application-row')
```

Comment:
not gonna block on this but it would be nice to look this up by role, if there's a role for table rows.

Reply:
Let me look into more after this PR.. :)